app/renderer/components/messageBox.js

```diff
@@ -0,0 +1,138 @@
+const commonStyles = require('./styles/commonStyles')
+
+class MessageBox extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onClick = this.onClick.bind(this)
+    this.onSuppressChanged = this.onSuppressChanged.bind(this)
+  }
+
+  get tabId () {
+    return this.props.tabId || ''
+  }
+
+  get title () {
+    const title = (this.props.detail && this.props.detail.get('title')) || ''
+    return title.replace(/^An embedded page at /, '')
+  }
+
+  get message () {
+    return (this.props.detail && this.props.detail.get('message')) || ''
+  }
+
+  get buttons () {
+    return (this.props.detail && this.props.detail.get('buttons')) || ['OK']
+  }
+
+  get cancelId () {
+    return this.props.detail && this.props.detail.get('cancelId')
+  }
+
+  get suppress () {
+    return (this.props.detail && this.props.detail.get('suppress')) || false
+  }
+
+  get showSuppress () {
+    return (this.props.detail && this.props.detail.get('showSuppress')) || false
+  }
+
+  onClick (e) {
+    e.stopPropagation()
+  }
+
+  onSuppressChanged (e) {
+    const detail = this.props.detail.toJS()
+    detail.suppress = !detail.suppress
+    appActions.updateMessageBox(this.tabId, detail)
+  }
+
+  onDismiss (tabId, buttonId) {
+    const response = {
+      suppress: this.suppress,
+      result: true
+    }
+
+    if (this.cancelId) {
+      response.result = buttonId !== this.cancelId
+    }
+
+    appActions.closeMessageBox(this.tabId, response)
+  }
+
+  get messageBoxButtons () {
+    const buttons = []
+
+    for (let index = (this.buttons.size - 1); index > -1; index--) {
+      buttons.push(<Button l10nId={this.buttons.get(index)}
+        className={index === 0 ? 'primaryButton' : 'whiteButton'}
+        onClick={this.onDismiss.bind(this, this.tabId, index)} />)
+    }
+
+    return buttons
+  }
+
+  render () {
+    return <Dialog className='noOutline'>
+      <div data-test-id={this.tabId}
+        onClick={this.onClick}
+        className={css(
+          commonStyles.flyoutDialog,
+          styles.container
+        )}>
+        <div className={css(styles.title)}>
+          {this.title}
+        </div>
+        {
+          this.showSuppress
+            ? <SwitchControl
+              rightl10nId='preventMoreAlerts'
+              checkedOn={this.suppress}
+              onClick={this.onSuppressChanged} />
+            : null
+        }
+        <div className={css(styles.body)}>
+          {this.message}
+        </div>
+        <div className={css(styles.buttons)}>
+          {this.messageBoxButtons}
+        </div>
+      </div>
+    </Dialog>
+  }
+}
+
+const styles = StyleSheet.create({
+  'container': {
+    margin: '3em',
+    outline: 'none',
+    display: 'block'
+  },
+  'title': {
+    fontWeight: 'bold',
+    fontSize: '12pt',
+    marginBottom: '1em',
+    marginTop: '0.5em'
+  },
+  'body': {
+    marginTop: '1.5em',
+    minWidth: '425px',
+    marginBottom: '1.5em'
+  },
+  'buttons': {
+    float: 'right',
```

Comment:
This works enough but I would not set with float to avoid the possibility of the case like this: https://github.com/brave/browser-laptop/issues/6749#issuecomment-274096915. I would instead use display: flex and justify-content: flex-end